rslib/src/search/sqlwriter.rs

```diff
@@ -490,18 +539,39 @@ impl SqlWriter<'_> {
     }
 
     fn write_single_field(&mut self, field_name: &str, val: &str) -> Result<()> {
-        let field_indicies_by_notetype = self.fields_indices_by_notetype(field_name)?;
+        let field_indicies_by_notetype =
+            self.num_fields_and_fields_indices_by_notetype(field_name)?;
         if field_indicies_by_notetype.is_empty() {
             write!(self.sql, "false").unwrap();
             return Ok(());
         }
 
         self.args.push(to_sql(val).into());
         let arg_idx = self.args.len();
+        let field_idx_str = format!("' || ?{arg_idx} || '");
+        let other_idx_str = "%".to_string();
+
+        let notetype_clause = |SingleFieldSearchContext {
+                                   mid,
+                                   num_fields,
+                                   fields,
+                               }: &SingleFieldSearchContext|
+         -> String {
+            let field_index_clause = |range: &Range<u32>| {
+                let f = (0..*num_fields)
+                    .filter_map(|i| {
+                        if i as u32 == range.start {
+                            Some(&field_idx_str)
+                        } else if range.contains(&(i as u32)) {
+                            None
+                        } else {
+                            Some(&other_idx_str)
+                        }
+                    })
+                    .join("\x1f");
+                format!("n.flds like '{f}' escape '\\'")
+            };
```

Comment:
I think I'm missing something here - if we ignore the exclusions for single field searches, why does this routine need to change?

Reply:
I thought maybe we want to keep using this approach since it's faster?

Reply:
Ah, that's what I was missing :-)